src/renderers/shared/reconciler/ReactNativeComponent.js

```diff
@@ -44,6 +45,38 @@ var ReactNativeComponentInjection = {
 };
 
 /**
+ * Wrap stateless function by component class
+
+ * @internal
+ * @param {function} Stateless component function
+ * @return {function} Component class
+ */
+function wrapStatelessFunction(fn) {
+  var Klass;
+
+  for (var i = 0; i < wrappedStatelessFunctions.length; i++) {
+    if (wrappedStatelessFunctions[i].fn === fn) {
+      Klass = wrappedStatelessFunctions[i].Klass;
+      break;
+    }
+  }
+
+  if (!Klass) {
+    Klass = function(props) {
+      return {
+        render: function() {
+          return fn(props);
+        }
+      };
+    };
+
+    wrappedStatelessFunctions.push({fn: fn, Klass: Klass});
```

Comment:
It's safer and faster to store the result as a property on the function (and use WeakMap when it becomes widely supported). It's not super sanitary, but pretty common JS-practice.